sale_line_cost_control/wizards/set_sale_line_purchase_price.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+import odoo.addons.decimal_precision as dp
+
+
+class SetSaleLinePurchasePrice(models.TransientModel):
+    _name = 'set.sale.line.purchase.price'
+
+    line_id = fields.Many2one(
+        comodel_name='sale.order.line',
+        string='Sale Line',
+        readonly=True,
+        required=True,
+    )
+    purchase_price = fields.Float(
+        string='Cost',
+        digits=dp.get_precision('Product Price'),
+        groups="base.group_user",
+        help="Cost of the product template used for standard stock valuation "
+             "in accounting and used as a base price on purchase orders. "
+             "Expressed in the default unit of measure of the product.")
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+        string="Currency",
+        required=True,
```

Comment:
Currently it's the sale order's currency. But now you tell it, I tend to think it makes more sense to use the company one...